src/python/engagement-creator/src/engagement-creator.py

```diff
@@ -76,59 +71,36 @@ def attach_risk(
     risk_node_uid = upsert(client, risk_node)
 
     create_edge(client, node_uid, "risks", risk_node_uid)
+    create_edge(client, risk_node_uid, "risky_nodes", node_uid)
+
+
+def recalculate_score(lens: LensView) -> int:
+    scope = lens.get_scope()
+    key_to_analyzers = defaultdict(set)
+    node_risk_scores = defaultdict(int)
+    total_risk_score = 0
+    for node in scope:
+        node_risks = node.get_risks()
+        risks_by_analyzer = {}
+        for risk in node_risks:
+            risk_score = risk.get_risk_score()
+            analyzer_name = risk.get_analyzer_name()
+            print(node.node_key, analyzer_name, risk_score)
+            risks_by_analyzer[analyzer_name] = risk_score
+            key_to_analyzers[node.node_key].add(analyzer_name)
+        node_risk_scores[node.node_key] = sum(
```

Comment:
bump

Reply:
We can either use a generator or calculate it once, not both, since once the generator is consumed it doesn't reset. I'll construct a list once, I suspect that's the happiest path here.

Reply:
This is what I meant:
```python
        value = sum(a for a in risks_by_analyzer.values() if a)
        node_risk_scores[node.node_key] = value
        total_risk_score += value
```